src/styles/blocks/catalog.scss

```diff
@@ -0,0 +1,7 @@
+.catalog {
+  padding: 50px 40px;
+  display: grid;
+  grid-template-columns: repeat(auto-fill, minmax(200px, 200px));
```

Comment:
The grid layout should adjust the number of columns based on specific screen resolutions. Consider using media queries to define different `grid-template-columns` for various breakpoints to meet the task requirements.